Enviando dados para o serviço de leads

GuilhermehChaves opened the pull request:
closes #3

GuilhermehChaves committed:
enviando dados

realizando a chamada da API para envio dos dados ao enviar o formulário

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,35 @@
+const form = document.getElementsByTagName("form")[0];
```
Algum motivo por fazer dessa forma ao invés de usar o jQuery que já está no site?

GuilhermehChaves replied:
Na verdade não.

DennyLoko replied:
Então acho melhor passar esse arquivo para usar jQuery... Já que já estamos incluindo o script, bora usar o que ele oferece...

GuilhermehChaves committed:
usando jquery

com base no code review, enviando dados com jquery

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,29 @@
+$("form").on("submit", function (event) {
```
Ta faltando fazer uma validação dos campos aqui... Bora validar se o telefone é válido e se os campos obrigatórios estão preenchidas, que são:
- Nome
- E-mail

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,29 @@
+
+    function onSuccess(response) {
+        if (response.status == "success") {
+            alert("Cadastro realizado com sucesso!");
```
O designer mandou o layout de como exibir os erros para o usuário e a mensagem de sucesso. Está lá no figma, no arquivo `landing 2.0`, as telas estão na parte à direita da página `Landing-page`. Ele fez os modelos mobile e desktop...

GuilhermehChaves committed:
realizando validações

com base no code review, realizando a validação dos campos do formulário

GuilhermehChaves committed:
exibindo mensagem

com base no code review, exibindo mensagem de sucesso e erro

DennyLoko commented on src/index.html:
```diff
@@ -307,11 +307,13 @@ <h1 class="title">
               <div class="register__form-content__form-group">
                 <label for="cellphone" class="subtitle">Celular</label>
                 <div class="border border--input">
-                  <input class="border__input" id="cellphone" type="tel" placeholder="(__) _____-____"
```
Precisou remover o espaço por causa do validador de telefone?

GuilhermehChaves replied:
Na verdade não, é porque no figma parecia que não tinha espaço, mas dando zoom da pra ver que realmente tem

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,80 @@
+$("form").on("submit", function (event) {
+    event.preventDefault();
+
+    function isEmpty(string) {
```
Está acontecendo uma coisa aqui que não é legal. Você está definindo essas funções toda vez que fazemos o submit do formulário. O ideal seria defini-las quando o DOM for carregado, pois assim não as define repetidas vezes...

Também acho que vai ficar mais organizado se você fizer uma classe contendo as validações e fazer o bind de algum método dela para ser executado ao enviar o formulário, mantendo tudo encapsulado...

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,80 @@
+
+    const inputs = $(".border--input");
+
+    if (isEmpty(body.name)) {
```
Você adicionou os erros ao enviar o formulário mas não está resetando quando o usuário manda enviar denovo, então a formatação do erro está ficando mesmo quando o campo não está incorreto...

GuilhermehChaves committed:
criando classe para validação

com base no code review, criando uma classe para fazer as validações do
formulário

GuilhermehChaves committed:
resetando input

com base no code review, resetando input quando há algum erro e o
usuário erá enviar o formulário novamente

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,122 @@
+            "email": {
+                input: $(this.form[0][1]),
+                border: $($(".border--input")[1]),
+                errorMessage: "informe o seu email.",
```
```suggestion
                errorMessage: "Informe o seu email.",
```

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,122 @@
+$("form").on("submit", function (event) {
+    event.preventDefault();
+    formControl.submit();
+});
```
Passa essa declaração para o construtor, pq ai fica tudo encapsulado na classe... Se passar o `preventDefault` para o `submit()`, você pode fazer algo assim:
```
this.form.on('submit', (e) => this.submit(e));
```

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,122 @@
+    if (formControl.inputs[this.name].error) {
+        formControl.reset(this);
+    }
+});
```
Recomendo mover isso aqui para a classe, e também, ao invés de usar o evento `focus` usar o `change`, pq ai só remove a formatação de erro se a pessoa preencher o input.

DennyLoko commented on src/js/form.js:
```diff
@@ -0,0 +1,122 @@
+    }
+}
+
+const formControl = new FormControl("form");
```
Se fizer as modificações que eu sugeri, nem precisamos mais declarar uma constante para armazenar o objeto do FormControl, não poluindo o namespace, podemos só instanciar que ele vai gerenciar bonitinho...

```suggestion
new FormControl("form");
```

DennyLoko reviewed (changes requested):
Nossa, usar essa classe ficou bem melhor! Ficou show! Só muda umas coisinhas para ficar mais bonito...

GuilhermehChaves committed:
adicionando on submit na classe

com base no code review, encapsulando on submit na class FormControl

GuilhermehChaves committed:
adicionando on change no construtor

com base no code review, encapsulando o on change na classe

GuilhermehChaves committed:
corrigindo mensagem

baseado no code review corrigindo mensagem de erro do email

GuilhermehChaves committed:
inserindo espaço no telefone

com base no code review, adicionando espaço no telefone
